cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -206,22 +206,19 @@ func (m *gceManagerImpl) registerMig(mig Mig) bool {
 
 // GetMigSize gets MIG size.
 func (m *gceManagerImpl) GetMigSize(mig Mig) (int64, error) {
-	if migSize, found := m.cache.GetMigTargetSize(mig.GceRef()); found {
-		return migSize, nil
-	}
-	targetSize, err := m.GceService.FetchMigTargetSize(mig.GceRef())
-	if err != nil {
-		return -1, err
-	}
-	m.cache.SetMigTargetSize(mig.GceRef(), targetSize)
-	return targetSize, nil
+	return m.cache.GetMigTargetSize(mig.GceRef())
 }
 
 // SetMigSize sets MIG size.
 func (m *gceManagerImpl) SetMigSize(mig Mig, size int64) error {
 	klog.V(0).Infof("Setting mig size %s to %d", mig.Id(), size)
 	m.cache.InvalidateTargetSizeCacheForMig(mig.GceRef())
-	return m.GceService.ResizeMig(mig.GceRef(), size)
+	err := m.GceService.ResizeMig(mig.GceRef(), size)
+	if err != nil {
+		return err
+	}
+	m.cache.SetMigTargetSize(mig.GceRef(), size)
```

Comment:
Do we need this, considering we invalidate this cache just before resizing?

Reply:
I guess it to handle the case when `ResizeMig` succeeded (changed the mig size on the server side) but returned error due to some network issues later on.
We do not want to have stale data in cache in that situation.

Reply:
But in that case you would still fetch target size of that mig next time you need it and not return stale data (since you invalidated it before ResizeMig call)?

Reply:
With invalidate call yes. Without we would return stale value from cache.

Reply:
After discussing with @losipiuk - yes it would be safe to skip this, but it's also safe to leave it and it potentially saves us a completely unnecessary API call (note we'll invalidate all target sizes for next loop anyway).

Ps. I think @losipiuk misunderstood original question :)